parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeometryStatistics.java

```diff
@@ -0,0 +1,168 @@
+
+  // Metadata that may impact the statistics calculation
+  private final LogicalTypeAnnotation.Edges edges;
+  private final String crs;
+  private final ByteBuffer metadata;
+
+  private final BoundingBox boundingBox;
+  private final List<Covering> coverings;
+  private final GeometryTypes geometryTypes;
+  private final WKBReader reader = new WKBReader();
+
+  public GeometryStatistics(
+      LogicalTypeAnnotation.Edges edges,
+      String crs,
+      ByteBuffer metadata,
+      BoundingBox boundingBox,
+      List<Covering> coverings,
+      GeometryTypes geometryTypes) {
+    this.edges = edges;
+    this.crs = crs;
+    this.metadata = metadata;
+    this.boundingBox = supportsBoundingBox() ? boundingBox : null;
+    this.coverings = supportsCovering() ? coverings : null;
+    this.geometryTypes = geometryTypes;
+  }
+
+  public GeometryStatistics(LogicalTypeAnnotation.Edges edges, String crs, ByteBuffer metadata) {
+    this(
+        edges,
+        crs,
+        metadata,
+        new BoundingBox(),
+        Arrays.asList(new EnvelopeCovering(edges, crs)),
+        new GeometryTypes());
+  }
+
+  public BoundingBox getBoundingBox() {
+    return boundingBox;
+  }
+
+  public List<Covering> getCoverings() {
+    return coverings;
+  }
+
+  public GeometryTypes getGeometryTypes() {
+    return geometryTypes;
+  }
+
+  public void update(Binary value) {
+    if (value == null) {
+      return;
+    }
+    try {
+      Geometry geom = reader.read(value.getBytes());
+      update(geom);
+    } catch (ParseException e) {
+      abort();
+    }
+  }
+
+  private void update(Geometry geom) {
+    if (supportsBoundingBox()) {
+      boundingBox.update(geom);
+    }
+    if (supportsCovering()) {
+      coverings.stream().forEach(c -> c.update(geom));
+    }
+    geometryTypes.update(geom);
+  }
+
+  /**
+   * A bounding box is a rectangular region defined by two points, the lower left
+   * and upper right corners. It is used to represent the minimum and maximum
+   * coordinates of a geometry. Only planar geometries can have a bounding box.
+   */
+  private boolean supportsBoundingBox() {
+    // Only planar geometries can have a bounding box
+    // based on the current specification
+    return edges == LogicalTypeAnnotation.Edges.PLANAR;
+  }
+
+  /**
+   * A custom WKB-encoded polygon or multi-polygon to represent a covering of
+   * geometries. For example, it may be a bounding box, or an envelope of geometries
+   * when a bounding box cannot be built (e.g. a geometry has spherical edges, or if
+   * an edge of geographic coordinates crosses the antimeridian). In addition, it can
+   * also be used to provide vendor-agnostic coverings like S2 or H3 grids.
+   */
+  private boolean supportsCovering() {
+    // This version (POC) assumes only build coverings for planar edges
+    // In case of spherical edges, no coverings are built
+    //     return edges == LogicalTypeAnnotation.Edges.PLANAR;
+    return true;
```

Comment:
As I have commented in the `EnvelopeCovering`, we'd better disable building covering for spherical edges. Therefore we cannot always return true here.